app/src/ui/diff/side-by-side-diff-row.tsx

```diff
@@ -406,7 +430,10 @@ export class SideBySideDiffRow extends React.Component<
                 after.isSelected
               )}
               {this.renderContent(after, DiffRowPrefix.Added)}
-              {this.renderWhitespaceHintPopover(DiffColumn.After)}
+              {this.props.renderWhitespaceHintPopover(
+                DiffColumn.After,
+                this.props.numRow
+              )}
```

Comment:
These calls to `this.props.renderWhitespaceHintPopover` are causing the problem @DylanDevelops 

These are changing the state of the parent element during render, instead of just rendering. I think the easiest approach is to keep the the rendering logic in the row based but based in a `showWhitespaceHint` prop instead of state.

Whenever that prop must change, the row can use a `this.props.onSomethingHappened` kind of method (please, choose a better name 😂 ) to inform the parent, and the parent will change the `showWhitespaceHint` prop of the row.

Does that make any sense? It's a bit of a convoluted approach but the price of having some logic in the row (line selection) and some logic in the parent (context menu) 😭